### ISSUE TITLE
[BUG] regexp_extract doesn't work correctly with concat 

### ISSUE THREAD

**Issue by sperlingxx** [NVIDIA/spark-rapids#5088]
**Describe the bug**
`regexp_extract(concat(....), )` produces incorrect results in GPU runs, when children of concat contains at least one column vector.

**Steps/Code to reproduce bug**
```scala
val df = (1 to 10).toDF("a")
spark.conf.set("spark.rapids.sql.regexp.enabled", "true")
df.coalesce(1).select(regexp_extract(concat(col("a"), lit("a")), "(a)", 1)).collect()
```
GPU results: `Array([], [], [], [], [], [], [], [], [], [])`
CPU results: `Array([a], [a], [a], [a], [a], [a], [a], [a], [a], [a])`

For above query, GPU works correctly only when `column(a)` outputs an empty string.

**Comment by revans2** [comment]
This has nothing to do with concat. It appears that `regexp_extract` on the CPU is doing a find, where as on the GPU it is doing a full match.
```
val df =Seq("1a", "2a", "3a", "4a", "5a", "6a", "7a", "8a", "9a", "10a").toDF("c")
df.coalesce(1).select(regexp_extract(col("c"), "(a)", 1)).collect()
```

shows the same results, but changing the regular expression to be ".*(a).*" produces the same result for both the CPU and the GPU.

**Comment by sperlingxx** [comment]
Close this issue since it is included in #5135
